locker/locking.go

```diff
@@ -0,0 +1,70 @@
+package locker
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/featureform/fferr"
+)
+
+const (
+	UpdateSleepTime = 2 * time.Second
+	ValidTimePeriod = 5 * time.Second
+)
+
+type LockInformation struct {
+	ID   string
+	Key  string
+	Date time.Time
+}
+
+func (l *LockInformation) Unmarshal(data []byte) fferr.GRPCError {
+	var tmp struct {
+		ID   string
+		Key  string
+		Date string
+	}
+	if err := json.Unmarshal(data, &tmp); err != nil {
+		return fferr.NewInternalError(err)
+	}
+
+	if tmp.ID == "" {
+		err := fmt.Errorf("lock information is missing ID")
+		return fferr.NewInvalidArgumentError(err)
+	}
+	if tmp.Key == "" {
+		err := fmt.Errorf("lock information is missing Key")
+		return fferr.NewInvalidArgumentError(err)
+	}
+
+	l.ID = tmp.ID
+	l.Key = tmp.Key
+
+	// Parse the date string with UTC time zone
+	parsedTime, err := time.Parse(time.RFC3339, tmp.Date)
+	if err != nil {
+		return fferr.NewInternalError(err)
```

Comment:
nit: I'd add context to the error like `id` and `key` errors above using addDetail
```
parseErr := fferr.NewInternalError(err)
parseErr.AddDetail("parseDate", "lock information cannot parse date field")
```
the regular parse error already has the attempted info, so just adding lock-specific context.

Reply:
maybe consider creating a new type. NewParsingError or something.